### PR TITLE
Rename dataset endpoints as asset endpoints

### PR DESCRIPTION
**Pull request by Lee-W**
* Rename dataset endpoints as asset endpoints
  * Rename ``/datasets`` as ``/assets``
  * Rename ``/datasets/{uri}`` as ``/assets/{uri}``
  * Rename ``/datasets/events`` as ``/assets/events``
  * Rename ``/datasets/queuedEvent/{uri}`` as ``/ui/next_run_assets/upstream``
  * Rename ``/dags/{dag_id}/dagRuns/{dag_run_id}/upstreamDatasetEvents`` as ``/ui/next_run_assets/upstream``
  * Rename ``/dags/{dag_id}/datasets/queuedEvent/{uri}`` as ``/ui/next_run_assets/upstream``
  * Rename ``/dags/{dag_id}/datasets/queuedEvent`` as ``/ui/next_run_assets/upstream``
  * Rename ``/ui/next_run_datasets/upstream`` as ``/ui/next_run_assets/upstream``
* Rename dataset schema as asset endpoints
  * Rename ``AssetCollection.datasets`` as ``AssetCollection.assets``
  * Rename ``AssetEventCollection.dataset_events`` as ``AssetEventCollection.asset_events``
  * Rename ``AssetEventCollectionSchema.dataset_events`` as ``AssetEventCollectionSchema.asset_events``
  * Rename ``CreateAssetEventSchema.dataset_uri`` as ``CreateAssetEventSchema.asset_uri``

---

Some of the names are still kept as Dataset due to the database not being changed. will be handled in https://github.com/apache/airflow/pull/42023

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_connexion/endpoints/asset_endpoint.py [hunk 1]
```diff
@@ -149,14 +149,13 @@ def get_dataset_events(
...
     return asset_event_collection_schema.dump(
-        AssetEventCollection(dataset_events=events, total_entries=total_entries)
+        AssetEventCollection(asset_events=events, total_entries=total_entries)
     )
 
 
 def _generate_queued_event_where_clause(
     *,
     dag_id: str | None = None,
-    dataset_id: int | None = None,
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/asset_endpoint.py:159]: Shouldn’t this now have an `asset_id`?

Comment (Lee-W) [reply airflow/api_connexion/endpoints/asset_endpoint.py:159]: yes, this one is missed. let me change it now.

Comment (Lee-W) [reply airflow/api_connexion/endpoints/asset_endpoint.py:159]: oh, this was not actually used. so that's why I removed it

Comment (uranusjr) [reply airflow/api_connexion/endpoints/asset_endpoint.py:159]: OK if it’s not used we don’t need to add it.

File: airflow/api_connexion/endpoints/asset_endpoint.py [hunk 2]
```diff
@@ -165,8 +164,6 @@ def _generate_queued_event_where_clause(
     where_clause = []
     if dag_id is not None:
         where_clause.append(AssetDagRunQueue.target_dag_id == dag_id)
-    if dataset_id is not None:
-        where_clause.append(AssetDagRunQueue.dataset_id == dataset_id)
```
Comment (vincbeck) [inline airflow/api_connexion/endpoints/asset_endpoint.py:169]: Similar to comment above. Should `AssetDagRunQueue` not have `asset_id` column and have a where clause similar to this one:

```
if asset_id is not None:
        where_clause.append(AssetDagRunQueue.asset_id == asset_id)
```

File: airflow/www/static/js/components/DatasetEventCard.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ import SourceTaskInstance from "./SourceTaskInstance";
 import TriggeredDagRuns from "./TriggeredDagRuns";
 
 type CardProps = {
-  datasetEvent: DatasetEvent;
+  datasetEvent: AssetEvent;
```
Comment (vincbeck) [inline airflow/www/static/js/components/DatasetEventCard.tsx]: Should we?

```suggestion
  assetEvent: AssetEvent;
```

File: airflow/www/static/js/components/SourceTaskInstance.tsx [hunk 4]
```diff
@@ -22,15 +22,15 @@ import { Box, Link, Tooltip, Flex } from "@chakra-ui/react";
...
+import type { AssetEvent } from "src/types/api-generated";
 import { useContainerRef } from "src/context/containerRef";
 import { SimpleStatus } from "src/dag/StatusBox";
 import InstanceTooltip from "src/components/InstanceTooltip";
 import type { TaskInstance } from "src/types";
 import { getMetaValue } from "src/utils";
 
 type SourceTIProps = {
-  datasetEvent: DatasetEvent;
+  datasetEvent: AssetEvent;
```
Comment (vincbeck) [inline airflow/www/static/js/components/SourceTaskInstance.tsx]: Should we?

```suggestion
  assetEvent: AssetEvent;
```

File: airflow/www/static/js/dag/details/graph/index.tsx [hunk 5]
```diff
@@ -150,7 +150,7 @@ const Graph = ({ openGroupIds, onToggleGroups, hoveredTaskState }: Props) => {
     setArrange(data?.arrange || "LR");
   }, [data?.arrange]);
 
-  const { data: datasetsCollection } = useDatasets({
+  const { data: datasetsCollection } = useAssets({
```
Comment (vincbeck) [inline airflow/www/static/js/dag/details/graph/index.tsx:153]: Same here but maybe you want to do it in a separate PR?

```suggestion
  const { data: assetsCollection } = useAssets({
```

File: airflow/www/static/js/dag/details/graph/Node.tsx [hunk 6]
```diff
@@ -49,7 +49,7 @@ export interface CustomNodeProps {
   style?: string;
   isZoomedOut: boolean;
   class: DepNode["value"]["class"];
-  datasetEvent?: DatasetEvent;
+  datasetEvent?: AssetEvent;
```
Comment (vincbeck) [inline airflow/www/static/js/dag/details/graph/Node.tsx]: Should we?

```suggestion
  assetEvent?: AssetEvent;
```

File: airflow/www/static/js/dag/details/graph/index.tsx [hunk 7]
```diff
@@ -185,33 +185,33 @@ const Graph = ({ openGroupIds, onToggleGroups, hoveredTaskState }: Props) => {
   if (showDatasets) {
     datasetNodes = [...upstreamDatasetNodes];
     datasetEdges = [...upstreamDatasetEdges];
-    datasetsCollection?.datasets?.forEach((dataset) => {
-      const producingTask = dataset?.producingTasks?.find(
+    datasetsCollection?.assets?.forEach((asset) => {
```
Comment (vincbeck) [inline airflow/www/static/js/dag/details/graph/index.tsx:188]:
```suggestion
    assetsCollection?.assets?.forEach((asset) => {
```

File: airflow/www/static/js/dag/details/graph/index.tsx [hunk 8]
```diff
@@ -185,33 +185,33 @@ const Graph = ({ openGroupIds, onToggleGroups, hoveredTaskState }: Props) => {
...
         }
       }
     });
 
     // Check if there is a dataset event even though we did not find a dataset
-    downstreamDatasetEvents.forEach((de) => {
+    downstreamAssetEvents.forEach((de) => {
       const hasNode = datasetNodes.find((node) => node.id === de.datasetUri);
       if (!hasNode && de.sourceTaskId && de.datasetUri) {
         datasetEdges.push({
```
Comment (vincbeck) [inline airflow/www/static/js/dag/details/graph/index.tsx:217]:
```suggestion
        assetEdges.push({
```

File: airflow/api_connexion/endpoints/asset_endpoint.py (hunk 2 above)
Comment (Lee-W) [reply airflow/api_connexion/endpoints/asset_endpoint.py:169]: Yes, that will be handled in https://github.com/apache/airflow/pull/42023. The db column has not yet changed

File: airflow/www/static/js/components/DatasetEventCard.tsx (hunk 3 above)
Comment (Lee-W) [reply airflow/www/static/js/components/DatasetEventCard.tsx]: Yes, this can be done in this PR.

File: airflow/www/static/js/components/SourceTaskInstance.tsx (hunk 4 above)
Comment (Lee-W) [reply airflow/www/static/js/components/SourceTaskInstance.tsx]: Yes, this can be done in this PR.

File: airflow/www/static/js/dag/details/graph/Node.tsx (hunk 6 above)
Comment (Lee-W) [reply airflow/www/static/js/dag/details/graph/Node.tsx]: Yes, this can be done in this PR.

File: airflow/www/static/js/dag/details/graph/index.tsx (hunk 5 above)
Comment (Lee-W) [reply airflow/www/static/js/dag/details/graph/index.tsx:153]: Yes, this will be included in the "UI file change "in https://github.com/apache/airflow/issues/42307

File: airflow/www/static/js/dag/details/graph/index.tsx (hunk 7 above)
Comment (Lee-W) [reply airflow/www/static/js/dag/details/graph/index.tsx:188]: this will be included in the "UI file change "in https://github.com/apache/airflow/issues/42307

File: airflow/www/static/js/dag/details/graph/index.tsx (hunk 8 above)
Comment (Lee-W) [reply airflow/www/static/js/dag/details/graph/index.tsx:217]: this will be included in the "UI file change "in https://github.com/apache/airflow/issues/42307

File: airflow/www/static/js/dag/details/graph/Node.tsx (hunk 6 above)
Comment (Lee-W) [reply airflow/www/static/js/dag/details/graph/Node.tsx]: Just updated

File: airflow/api_fastapi/views/ui/assets.py [hunk 9]
```diff
@@ -30,7 +30,7 @@
 assets_router = AirflowRouter(tags=["Asset"])
 
 
-@assets_router.get("/next_run_datasets/{dag_id}", include_in_schema=False)
+@assets_router.get("/next_run_assets/{dag_id}", include_in_schema=False)
```
Comment (bbovenzi) [inline airflow/api_fastapi/views/ui/assets.py:33]: Should we update the return data in this file too? `dataset_expression` -> `asset_expression`?

File: tests/api_fastapi/views/ui/test_assets.py [hunk 10]
```diff
@@ -34,14 +34,14 @@ def cleanup():
     initial_db_init()
 
 
-def test_next_run_datasets(test_client, dag_maker):
+def test_next_run_assets(test_client, dag_maker):
```
Comment (bbovenzi) [inline tests/api_fastapi/views/ui/test_assets.py:37]: Same comment about dataset_expression -> asset_expression here

File: airflow/www/static/js/api/useAsset.ts [hunk 11]
```diff
@@ -27,12 +27,12 @@ interface Props {
   uri: string;
 }
 
-export default function useDataset({ uri }: Props) {
+export default function useAsset({ uri }: Props) {
   return useQuery(["dataset", uri], () => {
     const datasetUrl = getMetaValue("dataset_api").replace(
```
Comment (bbovenzi) [inline airflow/www/static/js/api/useAsset.ts]:
```suggestion
    const datasetUrl = getMetaValue("asset_api").replace(
```
We changed the meta tag name

File: airflow/api_fastapi/views/ui/assets.py (hunk 9 above)
Comment (Lee-W) [reply airflow/api_fastapi/views/ui/assets.py:33]: This is also a db column in DAG model. I intend to change it in https://github.com/apache/airflow/pull/42023

File: tests/api_fastapi/views/ui/test_assets.py (hunk 10 above)
Comment (Lee-W) [reply tests/api_fastapi/views/ui/test_assets.py:37]: This is also a db column in DAG model. I intend to change it in https://github.com/apache/airflow/pull/42023

File: airflow/www/static/js/api/useAsset.ts (hunk 11 above)
Comment (Lee-W) [reply airflow/www/static/js/api/useAsset.ts]: This is something missed. Thanks for finding this. Just updated it

**Review by bbovenzi — APPROVED** [review]
Functionality lgtm

There's still variable names and places in the UI that use the name "dataset". But it is all deprecated so that's fine by me. Once we upgrade the db columns, let's make sure no mention of dataset finds its way into `/api_fastapi` and `/ui` directories.

**Comment by Lee-W** [comment]
> Functionality lgtm
> 
> There's still variable names and places in the UI that use the name "dataset". But it is all deprecated so that's fine by me. Once we upgrade the db columns, let's make sure no mention of dataset finds its way into `/api_fastapi` and `/ui` directories.

Sure! The next PR I'll work on is the DB one. once that's merged, I think we can get rid of the dataset in `/api_fastapi` and `/ui`. After that, I'll work on the rest of the UI thing, and everything has not yet changed

**Review by vincbeck — APPROVED** [review]

**Comment by jscheffl** [comment]
Assuming post-merge this PR broke canary tests on main - can you take a look for fixing? https://github.com/apache/airflow/actions/runs/11175856695/job/31068286744
